api/testutils/kusttest/harnessenhanced.go

```diff
@@ -109,12 +114,69 @@ func (th *HarnessEnhanced) LoadAndRunTransformer(
 	return resMap
 }
 
+func (th *HarnessEnhanced) TestTransformer(
+	config, input, expected string) {
+	for _, b := range []bool{true, false} {
+		th.t.Run(fmt.Sprintf("yaml-%v", b), func(t *testing.T) {
+			c, err := amendConfig(config, b)
+			if err != nil {
+				th.t.Fatalf("Err: %v", err)
+			}
+			resMap, err := th.RunTransformer(c, input)
+			if err != nil {
+				th.t.Fatalf("Err: %v", err)
+			}
+			th.AssertActualEqualsExpected(resMap, expected)
+		})
+	}
+}
+
+func amendConfig(config string, yamlSupport bool) (string, error) {
+	var out bytes.Buffer
+	rw := kio.ByteReadWriter{
+		Reader: bytes.NewBufferString(config),
+		Writer: &out,
+	}
+
+	err := kio.Pipeline{
+		Inputs: []kio.Reader{&rw},
+		Filters: []kio.Filter{
+			kio.FilterAll(yaml.FilterFunc(
+				func(node *yaml.RNode) (*yaml.RNode, error) {
+					return node.Pipe(yaml.FieldSetter{
+						Name:        "yamlSupport",
+						StringValue: strconv.FormatBool(yamlSupport),
+					})
+				}),
+			),
+		},
+		Outputs: []kio.Writer{&rw},
+	}.Execute()
+	return out.String(), err
+}
+
 func (th *HarnessEnhanced) ErrorFromLoadAndRunTransformer(
 	config, input string) error {
 	_, err := th.RunTransformer(config, input)
 	return err
 }
 
+type AssertFunc func(t *testing.T, err error)
+
+func (th *HarnessEnhanced) ErrorFromTestTransformer(
```

Comment:
how about: `RunTransformerAndHandleError`

Reply:
Updated the functions to be called `RunTransformerAndCheckError` and `RunTransformerAndCheckResult`.